File: 5-matematica/1279.cpp
#include <iostream>
#include <cstring>
// 
int main(){
    int v_a,e,par,impar,p,aap;
    std::string big;
    bool passou,orid;
    while(true){
        std::getline(std::cin,big);
        if(big=="\0"){
            break;
        }
        v_a=0;
        par=0;
        impar=0;
        e=1;
        passou=true;
        orid=false;
        for(auto i:big){
            v_a+=std::atoi(&i);
            (e%2==0)?par+=std::atoi(&i):impar+=std::atoi(&i);
            ++e;
        }
        aap=std::atoi(&big[big.length()-4]);
        p=std::atoi(&big[big.length()-2]);
        if(p%4==0&&aap%100){
            std::cout<<"This is leap year.\n";
            if(v_a%3==0){
                if(big[big.length()-1]=='0'|big[big.length()-1]=='5'){
                    std::cout<<"This is huluculu festival year.\n";
                }
                passou=false;
            }
            if(impar-par==0|(par-impar)%11==0){
                std::cout<<"This is bulukulu festival year.\n";
            }
            orid=true;
        }else if(aap%400==0){
            std::cout<<"This is leap year.\n";
            if(v_a%3==0){
                if(big[big.length()-1]=='0'|big[big.length()-1]=='5'){
                    std::cout<<"This is huluculu festival year.\n";
                }
                passou=false;
            }
            if(impar-par==0|(par-impar)%11==0){
                std::cout<<"This is bulukulu festival year.\n";
            }
            orid=true;
        }
        if(v_a%3==0&&passou){
            if(big[big.length()-1]=='0'|big[big.length()-1]=='5'){
                std::cout<<"This is huluculu festival year.\n";
                orid=true;
            }
        }
        if(!orid){
            std::cout<<"This is an ordinary year.\n";
        }
        big.clear();
        std::cout<<"\n";
    }
    return 0;
}

File: 5-matematica/1279.py
while True:
    try:
        x=input()
        v_a=0
        passou=True
        ordi=True
        e=1
        par=0
        inp=0
        for i in x:
            v_a+=int(i)
            if e%2!=0: inp+=int(i)
            else: par+=int(i)
            e+=1
        if int(x[len(x)-2]+x[len(x)-1])%4==0 and int(x[len(x)-4]+x[len(x)-3]+x[len(x)-2]+x[len(x)-1])%100!=0:
            print('This is leap year.')
            if v_a%3==0:
                if x[len(x)-1]=='5'or x[len(x)-1]=='0':
                    print('This is huluculu festival year.')
                    passou=False
            if par-inp==0 or (par-inp)%11==0:
                print('This is bulukulu festival year.')
            ordi=False
        elif int(x[len(x)-4]+x[len(x)-3]+x[len(x)-2]+x[len(x)-1])%400==0:
            print('This is leap year.')
            if v_a%3==0:
                if x[len(x)-1]=='5'or x[len(x)-1]=='0':
                    print('This is huluculu festival year.')
                    passou=False
            if par-inp==0 or (par-inp)%11==0:
                print('This is bulukulu festival year.')
            ordi=False
        if passou and v_a%3==0:
            if x[len(x)-1]=='5'or x[len(x)-1]=='0':
                print('This is huluculu festival year.')
                ordi=False
        if ordi:
            print('This is an ordinary year.')
        x=''
        print('\r')
    except: break